Remora.Commands/Builders/CommandBuilder.cs

```diff
@@ -0,0 +1,329 @@
+    public CommandBuilder(GroupBuilder? parent = null)
+    {
+        _name = string.Empty;
+        _aliases = new();
+        _attributes = new();
+        Parameters = new();
+        _conditions = new();
+
+        _parent = parent;
+
+        _parent?.Children.Add(this);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CommandBuilder"/> class.
+    /// </summary>
+    /// <param name="treeBuilder">The registration builder.</param>
+    public CommandBuilder(TreeRegistrationBuilder treeBuilder)
+        : this()
+    {
+        _treeBuilder = treeBuilder;
+    }
+
+    /// <summary>
+    /// Sets the name of the command.
+    /// </summary>
+    /// <param name="name">The name of the command.</param>
+    /// <returns>The current builder to chain calls with.</returns>
+    public CommandBuilder WithName(string name)
+    {
+        _name = name;
+        return this;
+    }
+
+    /// <summary>
+    /// Sets the description of the command.
+    /// </summary>
+    /// <param name="description">The description of the command.</param>
+    /// <returns>The current builder to chain calls with.</returns>
+    public CommandBuilder WithDescription(string description)
+    {
+        Description = description;
+        return this;
+    }
+
+    /// <summary>
+    /// Adds an alias to the command.
+    /// </summary>
+    /// <param name="alias">The alias to add.</param>
+    /// <returns>The current builder to chain calls with.</returns>
+    public CommandBuilder AddAlias(string alias)
+    {
+        _aliases.Add(alias);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds multiple aliases to the command.
+    /// </summary>
+    /// <param name="aliases">The aliases to add.</param>
+    /// <returns>The current builder to chain calls with.</returns>
+    public CommandBuilder AddAliases(IEnumerable<string> aliases)
+    {
+        _aliases.AddRange(aliases);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds an attribute to the command. Conditions must be added via <see cref="AddCondition"/>.
+    /// </summary>
+    /// <param name="attribute">The attribute to add.</param>
+    /// <returns>The current builder to chain calls with.</returns>
+    public CommandBuilder AddAttribute(Attribute attribute)
+    {
+        if (attribute is ConditionAttribute)
+        {
+            throw new InvalidOperationException("Conditions must be added via AddCondition.");
+        }
+
+        _attributes.Add(attribute);
+        return this;
+    }
+
+    /// <summary>
+    /// Adds a condition to the command.
+    /// </summary>
+    /// <param name="condition">The condition to add.</param>
+    /// <returns>The current builder to chain calls with.</returns>
+    public CommandBuilder AddCondition(ConditionAttribute condition)
+    {
+        _conditions.Add(condition);
+        return this;
+    }
+
+    /// <summary>
+    /// Applies a function in which the command is invoked by.
```

Comment:
This documentation needs a bit of work - it's kind of a word salad and I'm not sure what it means.

Reply:
Fair enough. I've reworded and will push when I resolve the rest of these comments